init.lua

```diff
@@ -283,12 +326,16 @@ minetest.register_node("warps:warpstone", {
 
 		local meta = minetest.get_meta(pos)
 		local destination = meta:get_string("warps_destination")
-		if destination == "" then
+		if destination == "" or lookup_warp(destination) == nil then
 			minetest.chat_send_player(puncher:get_player_name(),
 					"Unknown warp location for this warp stone, cannot warp!")
 			return false
 		end
+		minetest.log("action", string.format("Going to warp player %s to waypoint %s",
+			puncher:get_player_name(), minetest.formspec_escape(destination)
```

Comment:
`formspec_escape` is to escape strings for formspecs. this is a log message. no need to escape. It would only produce weird-looking output.

Reply:
I see that you now removed the `destination` parameter entirely, but this will result in unwanted behaviour because the second `%s` formatter/placeholder is not replaced with the destination waypoint.

Reply:
I swear luacheck did more harm than good